spec/models/user_spec.rb

```diff
@@ -1,5 +1,87 @@
-require 'rails_helper'
+require "rails_helper"
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe "enum for role" do
+    it {should define_enum_for(:role).with_values(admin: 0, customer: 1)}
+  end
+
+  describe "associations" do
+    it {should have_many(:orders).dependent(:destroy)}
+    it {should have_many(:comments).dependent(:destroy)}
+  end
+
+  describe "validates" do
```

Comment:
phân biệt describe, it, comtext khác gì nhau

Reply:
Describe thì nên dùng khi mô tả các method hoặc function được viết test. Còn context thì mô tả các trường hợp xảy ra đối với method hoặc function đó. It được sử dụng để định nghĩa một "Ví dụ" hay nói cách khác là dùng để định nghĩa 1 test case.